packages/adapter-components/src/fetch/element/type_utils.ts

```diff
@@ -379,3 +380,38 @@ export const removeNullValues = ({
     allowEmptyArrays,
     allowEmptyObjects,
   }) ?? {}
+
+/**
+ * Add empty object types for all types that can have instances in the workspace,
+ *  even if no instances will be created for them in the current fetch.
+ * This is needed because if instances are added / cloned from another environment,
+ *  they need to have a type in order to be deployed.
+ */
+export const createRemainingTypes = <Options extends FetchApiDefinitionsOptions>({
+  adapterName,
+  definedTypes,
+  defQuery,
+}: {
+  adapterName: string
+  definedTypes: Record<string, ObjectType>
+  defQuery: ElementAndResourceDefFinder<Options>
+}): Record<string, ObjectType> => {
+  const topLevelTypeNames = Object.keys(_.pickBy(defQuery.getAll(), def => def.element?.topLevel?.isTopLevel))
+  const missingTypes = topLevelTypeNames.filter(typeName => definedTypes[typeName] === undefined)
```

Comment:
nit: may be best to use 
```suggestion
  const missingTypes = topLevelTypeNames.filter(typeName => !Object.prototype.hasOwnProperty.call(definedTypes, typeName))
```
or alternatively 
```suggestion
  const missingTypes = topLevelTypeNames.filter(typeName => !isElement(definedTypes[typeName]))
```
to avoid failing on types such as `toString` (not very likely to be typenames and not fieldnames, but...)

Reply:
good point, thanks